### DISCUSSION TITLE
🏗️ sci-fi corridor

### DISCUSSION THREAD

**Discussion by lonnie-ralfs** [benevolent-games/website_2022#29] [Show and tell]
https://user-images.githubusercontent.com/19894550/152663830-1067db85-c4ef-4c43-9108-1d35cb85db47.mp4

**Comment by chase-moskal** [comment]
looks amazing!

- some of these graphical elements will really stretch the capabilities of our underlying engine, babylonjs
- it looks like volumetric light scattering will be possible! https://doc.babylonjs.com/divingDeeper/lights/volumetricLightScattering

also we need to be mindful of babylon's light limits, which is 4 per material. however, we can change the limit. as per [babylon docs](https://doc.babylonjs.com/divingDeeper/lights/lights_introduction#limitations), they note:

> Babylon.js allows you to create and register as many lights as you choose, but know that a single StandardMaterial can only handle a defined number simultaneous lights (by default this value is equal to 4 which means the first four enabled lights of the scene's lights list). You can change this number

they point out lights are hazardous to performance, especially on low-perf devices:

> But beware! Because with more dynamic lights, Babylon.js will generate bigger shaders which may not be compatible with low end devices like mobiles or small tablets. In this case, babylon.js will try to recompile shaders with less lights.
